pilot/pkg/kube/inject/inject.go

```diff
@@ -837,6 +838,10 @@ func intoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 
 	metadata.Annotations[annotation.SidecarStatus.Name] = status
 
+	if status != "" && mtlsReady && metadata.Annotations[annotation.AuthenticationMtlsReady.Name] == "" {
+		metadata.Annotations[annotation.AuthenticationMtlsReady.Name] = "true"
```

Comment:
>what's the expected behavior

I put the logic in there because I assumed that scenario would need to be covered in some regard.  Are you suggesting different behavior between scenario where annotations exists vs set explicitly to `""`?  Can golang even detect that case for map[string]string?